server/planning/planning/planning_service.py

```diff
@@ -0,0 +1,1568 @@
+            items = await results.to_list_raw()
+            results_count = len(items)
+
+            # If the total_items has not been set, then this is the first query
+            # In which case we need to store the total hits from the search
+            if total_items < 0:
+                total_items = results_count
+
+                # If the search doesn't contain any results, return here
+                if total_items < 1:
+                    break
+
+            # If the last query doesn't contain any results, return here
+            if results_count == 0:
+                break
+
+            total_received += results_count
+
+            # Yield the results for iteration by the callee
+            yield items
+
+    async def on_event_converted_to_recurring(self, updates: dict[str, Any], original: EventResourceModel):
+        for item in get_related_planning_for_events([original.id]):
+            related_events = get_related_event_links_for_planning(item)
+
+            # Set the ``recurrence_id`` in the ``planning.related_events`` field
+            for event in related_events:
+                if event["_id"] == original.id:
+                    event["recurrence_id"] = updates["recurrence_id"]
+                    break
+
+            await self.update(
+                item[ID_FIELD],
+                {
+                    "recurrence_id": updates["recurrence_id"],
+                    "related_events": related_events,
+                },
+            )
+
+    async def create(self, _docs: list[PlanningResourceModel | dict[str, Any]]) -> list[str]:
+        docs = await self._convert_dicts_to_model(_docs)
+        await self.prepare_planning_data(docs)
+        return await super().create(docs)
+
+    async def on_update(self, updates: dict[str, Any], original: PlanningResourceModel) -> None:
+        await super().on_update(updates, original)
+
+        update_method = updates.pop("update_method", UpdateMethods.SINGLE)
+        updates.setdefault("versioncreated", utcnow())
+
+        user = get_user()
+        self.validate_on_update(updates, original, user)
+
+        if user and user.get(ID_FIELD):
+            updates["version_creator"] = user[ID_FIELD]
+
+        updated_planning = PlanningResourceModel.from_dict(updates)
+        self._handle_coverages(updated_planning, original)
+        self.set_planning_schedule(updated_planning, original)
+
+        # set the updates dictionary with the values from the pydantic model
+        # in case the object was edited in-place.
+        # TODO-ASYNC: unify everything to use only pydantic models
+        updates = updated_planning.to_dict()
+
+        if update_method and update_method != UpdateMethods.SINGLE:
+            await self._update_recurring_planning_items(updates, original, update_method)
+
+    async def on_updated(self, updates: dict[str, Any], original_obj: PlanningResourceModel):
+        # TODO-ASYNC: figure out what to do with the `from_ingest` param here
```

Comment:
FYI: `from_ingest` is used from the [patch_in_mongo](https://github.com/superdesk/superdesk-planning/blob/develop/server/planning/planning/planning.py#L106), which is executed from an ingest cycle.

We will need to look at this once we work on the async ingest